pkg/print/common/errors.go

```diff
@@ -0,0 +1,34 @@
+// Copyright 2021 The Kubernetes Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package common
+
+import "fmt"
+
+// ResultError is returned from printers when the apply/destroy operations completed, but one or
+// more resources either failed apply/prune/delete, or failed to reconcile.
+type ResultError struct {
+	ApplyFailed  int
+	PruneFailed  int
+	DeleteFailed int
+
+	ReconcileTimeout int
+	ReconcileFailed  int
+}
+
+func (a *ResultError) Error() string {
+	failedActuateSum := a.ApplyFailed + a.PruneFailed + a.DeleteFailed
+	failedReconcileSum := a.ReconcileTimeout + a.ReconcileFailed
+	switch {
+	case failedActuateSum > 0 && failedReconcileSum > 0:
+		return fmt.Sprintf("%d resources failed, %d resources failed to reconcile before timeout",
+			failedActuateSum, failedReconcileSum)
+	case failedActuateSum > 0:
+		return fmt.Sprintf("%d resources failed", failedActuateSum)
+	case failedReconcileSum > 0:
+		return fmt.Sprintf("%d resources failed to reconcile before timeout",
+			failedReconcileSum)
+	default:
+		return "actuation and reconcile failed"
```

Comment:
The `default` case should mean that all the resources are reconciled, right?

Reply:
The default case for an error is still an error. Tho I don't know what it would mean in this context... seems invalid/unreachable.